qcodes/instrument/abstract.py

```diff
@@ -0,0 +1,96 @@
+from typing import Any, Type
+from qcodes import Parameter
+from qcodes.instrument.base import InstrumentBase
+from qcodes.instrument.parameter import _BaseParameter
+
+
+class AbstractParameter(Parameter):
+    """
+    This is a trivial subclass of 'Parameter' to signal
+    that this parameters *must* be overridden in
+    subclasses of abstract instruments.
+    """
```

Comment:
I don't think it is a good idea to make express the abstract. How would you mix this with an ArrayParameter, MultiParameter or any other class of parameter. It would be better to let abstract be a property on the BaseParameter class which can be set by passing an argument to add parameter. Furthermore this is consistent with how it is implemented on an Abstract Method